### ISSUE TITLE
Failed Building Environment using Conda

### ISSUE THREAD

**Issue by lazyoracle** [esowc/wildfire-forecasting#6]
## What

Failed to build environment using the `environment.yml` recipe.

## How

```bash
git clone https://github.com/esowc/wildfire-forecasting
cd wildfire-forecasting
conda env create --file environment.yml
```

## Errors

```bash
Collecting package metadata (repodata.json): done
Solving environment: failed

ResolvePackageNotFound:
  - libuuid=1.0.3
  - glib=2.63.1
  - libgcc-ng=9.1.0
  - gst-plugins-base=1.14.0
  - readline=8.0
  - gmp=6.1.2
  - graphviz=2.40.1
  - expat=2.2.6
  - libgfortran-ng=7.3.0
  - ncurses=6.2
  - harfbuzz=1.8.8
  - jasper=1.900.1
  - gstreamer=1.14.0
  - dbus=1.13.14
  - libedit=3.1.20181209
  - libstdcxx-ng=9.1.0
  - ld_impl_linux-64=2.33.1
```

## System Information

* OS - `Windows 10 Pro 2004 build 19041.329`
* Conda - `conda 4.8.3`
* Python - `Python 3.7.7`

**Comment by lazyoracle** [comment]
## Reproduce the issue on Binder

## Error log from Binder 

```bash
Preparing transaction: ...working... done
Verifying transaction: ...working... done
Executing transaction: ...working... b'Uninstalling jupyter-js-widgets jupyter-js-widgets/extension\nRemoving: /srv/conda/envs/notebook/share/jupyter/nbextensions/jupyter-js-widgets\n'
done
Ran pip subprocess with arguments:
['/srv/conda/envs/notebook/bin/python', '-m', 'pip', 'install', '-U', '-r', '/home/jovyan/condaenv.5a_d0do6.requirements.txt']
Pip subprocess output:
Collecting absl-py==0.9.0
  Downloading absl-py-0.9.0.tar.gz (104 kB)
Collecting affine==2.3.0
  Downloading affine-2.3.0-py2.py3-none-any.whl (15 kB)



==> WARNING: A newer version of conda exists. <==
  current version: 4.8.2
  latest version: 4.8.3

Please update conda by running

    $ conda update -n base conda


Pip subprocess error:
ERROR: Could not find a version that satisfies the requirement apex==0.1 (from -r /home/jovyan/condaenv.5a_d0do6.requirements.txt (line 3)) (from versions: 0.9.8dev.linux-i686, 0.9.8.dev0, 0.9.8a0.dev0, 0.9.9.dev0, 0.9.10.dev0)
ERROR: No matching distribution found for apex==0.1 (from -r /home/jovyan/condaenv.5a_d0do6.requirements.txt (line 3))


CondaEnvException: Pip failed

Removing intermediate container 9d2d5fecd233
The command '/bin/sh -c conda env update -p ${NB_PYTHON_PREFIX} -f "environment.yml" && conda clean --all -f -y && conda list -p ${NB_PYTHON_PREFIX}' returned a non-zero code: 1
```

**Comment by lazyoracle** [comment]
## Reproduce the issue on Ubuntu 20.04

## Error Log

```bash
Preparing transaction: done
Verifying transaction: done
Executing transaction: done
Ran pip subprocess with arguments:
['/home/anurag/miniconda3/envs/wildfire-dl/bin/python', '-m', 'pip', 'install', '-U', '-r', '/mnt/c/Users/Anurag/wildfire-forecasting/condaenv.1qi47mik.requirements.txt']
Pip subprocess output:
Collecting absl-py==0.9.0
  Downloading absl-py-0.9.0.tar.gz (104 kB)
Collecting affine==2.3.0
  Downloading affine-2.3.0-py2.py3-none-any.whl (15 kB)

Pip subprocess error:
ERROR: Could not find a version that satisfies the requirement apex==0.1 (from -r /mnt/c/Users/Anurag/wildfire-forecasting/condaenv.1qi47mik.requirements.txt (line 3)) (from versions: 0.9.8dev.linux-i686, 0.9.8.dev0, 0.9.8a0.dev0, 0.9.9.dev0, 0.9.10.dev0)
ERROR: No matching distribution found for apex==0.1 (from -r /mnt/c/Users/Anurag/wildfire-forecasting/condaenv.1qi47mik.requirements.txt (line 3))


CondaEnvException: Pip failed
```

## System Information

* OS - `Ubuntu 20.04 on WSL 2`
* Conda - `conda 4.8.3`
* Python - `Python 3.7.7`

**Comment by lazyoracle** [comment]
### Attempting to fix on Ubuntu 20.04 WSL 2
- Copy pip requirements from conda recipe to a `requirements.txt` file
- Running `pip install -r requirements.txt` produces errors for `apex` and `pytorch-lightning`: 
```bash
ERROR: Could not find a version that satisfies the requirement apex==0.1
------------------
ERROR: Could not find a version that satisfies the requirement pytorch-lightning==0.8.0.dev0
```
- Changed `apex==0.1` to `apex` and `pytorch-lightning==0.8.0.dev0` to `pytorch_lightning-0.8.1`

- Retrying to install using `pip` now completes the installation with the following error:
```bash
ERROR: sphinx-autobuild 0.7.1 has requirement port-for==0.3.1, but you'll have port-for 0.4 which is incompatible.
```

@roshni-b Any insights?
